product_packaging_container_deposit/models/container_deposit_order_mixin.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class OrderMixin(models.AbstractModel):
+    """This mixin should only be inherited by purchase.order and sale.order models."""
+
+    _name = "container.deposit.order.mixin"
+    _description = "Container Deposit Order Mixin"
+
+    def prepare_deposit_container_line(self, product, qty):
+        self.ensure_one()
+        values = {
+            "name": product.name,
+            "product_id": product.id,
+            self.order_line._get_product_qty_field(): qty,
+            "is_container_deposit": True,
+            "order_id": self.id,
+        }
+        return values
+
+    def update_order_container_deposit_quantity(self):
+        self = self.with_context(updating_container_deposit=True)
+        for order in self:
+            # Lines to compute container deposit
+            lines_to_comp_deposit = order.order_line.filtered(
```

Comment:
lines must be taken w/ a specific method that should be overridden by the extending module
```suggestion
            lines_to_comp_deposit = order[self._get_order_line_field()].filtered(
```

Reply:
I understand the principle,  however `order_line` it will be always the same as this module is supposed to be used only on purchase and sale.
But anyway I'll implement with `_get_order_line_field`

Reply:
done